### PR TITLE
kuz-610 - Implement regexp in dsl

### PR DESCRIPTION
**Pull request by benoitvidis**

### CODE REVIEW

**Comment by jenow** [comment]
+1

File: lib/api/dsl/methods.js [hunk 1]
```diff
@@ -670,8 +670,53 @@ module.exports = function Methods(filtersObject) {
...
+      flags;
+
+    if (typeof filter !== 'object') {
+      return q.reject(new BadRequestError('Regexp argument must be an object'));
+    }
+
+    for (k in filter) {
+      if (filter.hasOwnProperty(k)) {
+        field = k;
+        value = filter[k];
```
Comment (scottinet) [inline lib/api/dsl/methods.js]: If I understand correctly, this means that if there are multiple regular expressions defined, we silently take and process the last one.
Shouldn't it be safer to detect multiple regexp defined and throw a `BadRequestError`?

**Comment by scottinet** [comment]
+1

**Comment by dbengsch** [comment]
+1

**Pull request merged**